### DISCUSSION TITLE
form.reset for unmounted form fields not working as expected

### DISCUSSION THREAD

**Discussion by andrewmtam** [react-hook-form/react-hook-form#9086] [Q&A]
Hi all, I'm having trouble getting forms to "clear" using the `reset` call.  I know we can address this by setting `defaultValues` when we call `useForm`, but some of the forms we create are very large, and if we can find a way around it that would be appreciated.

Here's a sandbox to demonstrate what I'm talking about:
https://codesandbox.io/s/react-hook-form-js-forked-g7m718?file=/src/App.js

There are two behaviors that I'd like to demonstrate here.

Working scenario:
* Start with inputs unmounted (don't click "ToggleForm")
* Click "Reset"
* Click "ToggleForm"
    * Input should be empty
* Click "Reset with value"
    * Input should populate
* Click "Reset"
    * Input should clear

Broken scenario:
* Start with inputs unmounted (don't click "ToggleForm")
* Click "Reset with value"
* Click "ToggleForm"
    * Input should be populated
* Click "Reset"
    * Input should incorrectly stay populated

From these examples, things work as expected when we start with an empty form, but they do not work as expected when we start with a populated form.  Any guidance here would be appreciated, thank you!

**Comment by bluebill1049** [comment]
> but some of the forms we create are very large,

curious about how big are they? any chance to break them into forms? I am asking that because I see a lot of difficulties due to extra large forms without breaking them down into small sections. I can see your problem is due to inline `defaultVlaue`, maybe you want to provide updated inline defaultValue after partial reset?

**Reply by andrewmtam** [reply]
> curious about how big are they? any chance to break them into forms? I am asking that because I see a lot of difficulties due to extra large forms without breaking them down into small sections.

They can be anywhere from 10 - 20 fields (or more), including some nested field paths, for instance, something like this:
```
{
    firstName: ''.
    lastName: '',
    personalDetails: '',
    ...
    employment: {
        managerFirstName: '',
        managerLastName: ""
        role: ""
        ....
    }
}
```

It would be easiest if we can get away with inlining the `defaultValue` on the Controller as opposed to calling `useForm({ defaultValues: {...}})`

Can you elaborate on what you mean by this:
> I can see your problem is due to inline defaultVlaue, maybe you want to provide updated inline defaultValue after partial reset?

Is there a simple adjustment you'd be able to make to the codesandbox to make things work as expected?

Appreciate your time, thanks!

**Comment by leapful** [comment]
@andrewmtam : If "reset with value" button is where your logic performs then just reset the target default value before performing "reset with value".

Working demo: https://codesandbox.io/s/react-hook-form-js-forked-pjwpm5?file=/src/App.js
